### PR TITLE
Add extra explicit loaded class for filesystem

### PR DESCRIPTION
**Pull request by yyongycy**
### What changes are proposed in this pull request?

Add extra explicit loaded class for filesystem

### Why are the changes needed?

In some situation, it is expected to load a class explicit, so that the extended logic can be appended.

### Does this PR introduce any user facing changes?
NA

### CODE REVIEW

**Comment by yyongycy** [comment]
@Xenorith  This is a potential one required for branch-3.0-preview

**Review by JiamingMai — APPROVED** [review]
LGTM

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Comment by yyongycy** [comment]
alluxio-bot, cherry pick this to branch-3.0-preview

**Comment by yyongycy** [comment]
alluxio-bot, cherry pick this to branch-3.0-preview please

File: dora/core/client/fs/src/main/java/alluxio/client/file/FileSystem.java [hunk 1]
```diff
@@ -92,6 +92,14 @@ public interface FileSystem extends Closeable {
    * {@link Factory#create} methods will always guarantee returning a new FileSystem.
    */
   class Factory {
+
+    static {
+      // If the extra loaded class name is set, try to load it.
+      if (Configuration.global().isSet(PropertyKey.EXTRA_LOADED_FILESYSTEM_CLASSNAME)) {
+        Configuration.global().getClass(PropertyKey.EXTRA_LOADED_FILESYSTEM_CLASSNAME);
```
Comment (Jackson-Wang-7) [inline dora/core/client/fs/src/main/java/alluxio/client/file/FileSystem.java:99]: It's enough to just get the class name? Or do we need to load it?

Comment (yyongycy) [reply dora/core/client/fs/src/main/java/alluxio/client/file/FileSystem.java:99]: this getClass will load the class as well

  public <T> Class<T> getClass(PropertyKey key) {
    Object value = get(key);
    if (value instanceof Class) {
      return (Class<T>) value;
    }
    try {
      return (Class<T>) Class.forName((String) value);
    } catch (ClassNotFoundException e) {
      throw new IllegalStateException(
          format("Requested class %s can not be loaded", value));
    }
  }

**Pull request merged**